pkg/nsx/services/securitypolicy/convert.go

```diff
@@ -0,0 +1,285 @@
+package securitypolicy
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	crdv1alpha1 "github.com/vmware-tanzu/nsx-operator/pkg/apis/crd.nsx.vmware.com/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+func DeepCopyInto(in *v1alpha1.SecurityPolicy, out *crdv1alpha1.SecurityPolicy) {
```

Comment:
Did you write these manually? deepcopy-gen can be used to generate such code and is safer to use it to avoid unexpected copy issues.

Reply:
Yes, I manually updated some code based on deepcopy-gen, since it only supports the same type copy.

Reply:
I'm not confident on manual code on this and people would just forgot updating this file when the struct changes. I'm not sure what you mean "it only supports the same type copy", I believe it supports deepcopy of different types of same structs, which is its main use case.

Reply:
I tried conversion-gen, but it is a bit unfriendly to the user. I think twice since the structure is the same, only in a different package, it is safe to use 
```
func T1ToVPC(in *v1alpha1.SecurityPolicy, out *crdv1alpha1.SecurityPolicy) {
	out = (*crdv1alpha1.SecurityPolicy)(unsafe.Pointer(in))
	out.APIVersion = "crd.nsx.vmware.com/v1alpha1"
	controllerutil.RemoveFinalizer(out, common.T1SecurityPolicyFinalizerName)
	controllerutil.AddFinalizer(out, common.SecurityPolicyFinalizerName)
}

func VPCToT1(in *crdv1alpha1.SecurityPolicy, out *v1alpha1.SecurityPolicy) {
	out = (*v1alpha1.SecurityPolicy)(unsafe.Pointer(in))
	out.APIVersion = "nsx.vmware.com/v1alpha1"
	controllerutil.RemoveFinalizer(out, common.SecurityPolicyFinalizerName)
	controllerutil.AddFinalizer(out, common.T1SecurityPolicyFinalizerName)
}

```
and it is more concise for us, what do you think?